solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1239 @@
+		if (!engine)
+		{
+			serr() << "Invalid option for --" << g_argModelCheckerEngine << ": " << engineStr << endl;
+			return false;
+		}
+		m_options.modelCheckerSettings.engine = *engine;
+	}
+
+	if (m_args.count(g_argModelCheckerTargets))
+	{
+		string targetsStr = m_args[g_argModelCheckerTargets].as<string>();
+		optional<ModelCheckerTargets> targets = ModelCheckerTargets::fromString(targetsStr);
+		if (!targets)
+		{
+			serr() << "Invalid option for --" << g_argModelCheckerTargets << ": " << targetsStr << endl;
+			return false;
+		}
+		m_options.modelCheckerSettings.targets = *targets;
+	}
+
+	if (m_args.count(g_argModelCheckerTimeout))
+		m_options.modelCheckerSettings.timeout = m_args[g_argModelCheckerTimeout].as<unsigned>();
+
+	m_options.metadataLiteral = (m_args.count(g_argMetadataLiteral) > 0);
+	m_options.initializeModelChecker =
+		m_args.count(g_argModelCheckerContracts) ||
+		m_args.count(g_argModelCheckerEngine) ||
+		m_args.count(g_argModelCheckerTargets) ||
+		m_args.count(g_argModelCheckerTimeout);
+	m_options.experimentalViaIR = (m_args.count(g_argExperimentalViaIR) > 0);
+	m_options.expectedExecutionsPerDeployment = m_args[g_argOptimizeRuns].as<unsigned>();
+
+	m_options.optimize = (m_args.count(g_argOptimize) > 0);
+	m_options.noOptimizeYul = (m_args.count(g_strNoOptimizeYul) > 0);
+
+	OptimiserSettings settings = m_options.optimize ? OptimiserSettings::standard() : OptimiserSettings::minimal();
+	if (m_options.noOptimizeYul)
+		settings.runYulOptimiser = false;
+	if (m_args.count(g_strYulOptimizations))
+	{
+		if (!settings.runYulOptimiser)
+		{
+			serr() << "--" << g_strYulOptimizations << " is invalid if Yul optimizer is disabled" << endl;
+			return false;
+		}
+
+		try
+		{
+			yul::OptimiserSuite::validateSequence(m_args[g_strYulOptimizations].as<string>());
+		}
+		catch (yul::OptimizerException const& _exception)
+		{
+			serr() << "Invalid optimizer step sequence in --" << g_strYulOptimizations << ": " << _exception.what() << endl;
+			return false;
+		}
+
+		m_options.yulOptimiserSteps = m_args[g_strYulOptimizations].as<string>();
+	}
+
+	if (m_args.count(g_argImportAst) > 0)
+		m_options.inputMode = InputMode::CompilerWithASTImport;
+	else
+		m_options.errorRecovery = (m_args.count(g_argErrorRecovery) > 0);
+
+	solAssert(m_options.inputMode == InputMode::Compiler || m_options.inputMode == InputMode::CompilerWithASTImport, "");
+	return true;
+}
+
+bool CommandLineParser::hasOutput()
+{
+	return g_hasOutput;
+}
+
+bool CommandLineParser::parseCombinedJsonOption()
+{
+	if (!m_args.count(g_argCombinedJson))
+		return true;
+
+	set<string> requests;
+	for (string const& item: boost::split(requests, m_args[g_argCombinedJson].as<string>(), boost::is_any_of(",")))
+		if (!g_combinedJsonArgs.count(item))
+		{
+			serr() << "Invalid option to --" << g_argCombinedJson << ": " << item << endl;
+			return false;
+		}
+
+	m_options.combinedJsonRequests = CombinedJsonRequests{};
```

Comment:
```suggestion
	CombinedJsonRequests requests{};
	reuqests.abi = requests.count(g_strAbi) > 0;
	// ...
	m_options.combinedJsonRequests = move(requests);
```
I *think* this would remove quite some boilerplate from the source code that looks a little bulky otherwise (just a minor though).

Reply:
Not sure about this one. I'd rather leave it as is.
- The change to a local variable actually makes it longer (adds one line for the move)
- I'd rather leave the parentheses around the condition. I always put them around conditions because otherwise it looks ambiguous to me otherwise. I know comparisons have higher precendence than `=` but it still trips me up when I look at it.